content/200-concepts/100-components/01-prisma-schema/06-relations/420-referential-integrity.mdx

```diff
@@ -8,21 +8,46 @@ preview: true
+Referential integrity is a property of a data set that states that all its references are valid. Referential integrity requires that if one record references another, then the referenced record must exist. For example, if a `Post` model defines an author, then the author must also exist. Referential integrity is enforced through constraints that prevent you making changes that break references, and through defining [referential actions](/concepts/components/prisma-schema/relations/referential-actions) that run when updating or deleting records. In the case of the `Post` model, for example, you must define what happens if its author is deleted.
 
-The referential integrity can either be handled in the database using foreign keys, or in the Prisma Client. There are pros and cons on both. In general, if the database supports foreign keys, it is usually the preferred choice and the default for SQL databases when using Prisma.
+In Prisma, referential integrity is implemented by defining relations between records using the [`@relation`](/reference/api-reference/prisma-schema-reference#relation) attribute. The `onUpdate` and `onDelete` arguments of this attribute can be used to define which referential action take place on updating or deleting the related record.
 
-In certain cases, either when the database doesn't support foreign keys, or when they are not preferred due to scaling reasons, Prisma can disable them and handle the referential integrity in the client.
+For example, take the following blog post schema, with a one-to-many relation between `Post`s and `User`s:
+
+```prisma file=schema.prisma highlight=4;normal
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+In this case, the `onDelete: Cascade` argument specifies the `Cascade` referential action, deleting the `Post` if the related `User` is deleted. See [Referential actions](/concepts/components/prisma-schema/relations/referential-actions) for further details of supported actions.
+
+When using a relational database that supports foreign keys, referential integrity can be handled in the underlying database. This is the default option in Prisma when using SQL databases.
+
+Alternatively, Prisma can emulate referential integrity in the client. This is the default option when using MongoDB as a database, because it does not support foreign keys. Emulation can also be selected for relational databases in cases where foreign keys are not supported, or where it is preferred for other reasons (for example, in cases where there are scaling concerns).
```

Comment:
```suggestion
Alternatively, Prisma can emulate referential integrity in the client. This is the default option when using MongoDB as a database, because it does not support foreign keys. Emulation can also be enabled for relational databases in cases where foreign keys are not supported, or where it is preferred for other reasons (for example, in cases where there are scaling concerns).
```